src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -69,15 +69,17 @@ public class Session {
             CommandLineArguments = commandLineArguments;
         }
 
-        public void StartHost(SecureString password, string profilePath, ILogger outputLogger, ILogger messageLogger) {
+        public void StartHost(ILogger logger, SecureString password, string profilePath, ILogger outputLogger, ILogger messageLogger) {
```

Comment:
It's better to pass this logger to constructor of `Session` instead, and save it in a field. The reason why the other two loggers here are passed to `StartHost` is because the stuff they log corresponds to one particular host process (its error output and its message pipe end, respectively). So the lifetime of those log writers also corresponds to that of the host, or at least the caller can make it so by passing new instances every time.

Login errors, however, pertain to the user and session, not the host per se (since they happen before the host process is even started). So the logger where they go belongs to the `Session` instance, and shared by other logic in this class.